### ISSUE TITLE
Can i pair ?

### ISSUE THREAD

**Issue by tanhsnkt1997** [Linch1/Web3TokenCharting#1]
Can i pair only token/btc, token/usdt, token/doge, token/eth, token/usdc ? i wan't same poocoin.app =)

**Comment by tanhsnkt1997** [comment]
And what is _hexadem ? thanks you

**Comment by Linch1** [comment]
Ehi, you can have any pair that exists for that token, but I don't really get what you are asking with the first question, can you rephrase it ? 

_hexadem is a code inside the dex smart contracts that ( if I understood well by the code ) is used just for creating the unique pair addresses, i can think it like a dex 'sign' for it's pairs address.

**Comment by tanhsnkt1997** [comment]
![image](https://user-images.githubusercontent.com/45147625/152624339-bac751b3-4041-4e25-8b44-ed339574a8cb.png)
sometimes it gives error Error: Invalid JSON RPC response: "" as shown, do you know why?

**Comment by tanhsnkt1997** [comment]
![image](https://user-images.githubusercontent.com/45147625/152624460-58e45acc-43ba-4d66-a763-78b7f9a060fd.png)
And open, high, low, value always equal , only close different ?

**Comment by Linch1** [comment]
`JSON RPC response` is an issue given from your provider, you should detect the line where it raise the error and add a `try catch`, i missed some of them, if you find where it gives the error please make a pull also in the repo.

The open, high, low, value always equal for some tokens probably becouse during 1 minute only 1 transaction is made so the minute candle is not update except for the `close` field. the `close` field is always updated during the next transaction that is made with that token also if the transaction is made hours later.

If you see all the prices retrived for a single pair you should see that the `close` is always equal to the next price `open` , but for now i saw that there are still some issues in updating the `close` field, i'm working on that when i have time

**Comment by tanhsnkt1997** [comment]
![image](https://user-images.githubusercontent.com/45147625/152665429-52aafb20-1c6e-4c2e-99b1-834e7b51a799.png)
And token basic some tokens do not exist for example Ethereum Token (0x2170ed0880ac9a755fd29b2688956bd959f933f8), and as shown in the picture Tether USD appears many records I think it only has 1 record, total_supply and pairs_count will be updated when changed?

**Comment by Linch1** [comment]
Some tokens don't exists probably becouse the address is a non standard erc20, this cause an error when retriving the informations like `decimals` , `supply` etc ...

`0x2170ed0880ac9a755fd29b2688956bd959f933f8` this seems to be an address and not a token, am i wrong ?

In the token basic records the `pairs_count` are updated yes, the `total_supply` instead is never updated.

**Comment by tanhsnkt1997** [comment]
thanks for your explanation, can you help me how to calculate volume/1 block?

**Comment by tanhsnkt1997** [comment]
Or calculate trading volume of a tx ?
